pandas/core/groupby/groupby.py

```diff
@@ -1460,8 +1460,138 @@ def describe(self, **kwargs):
     @Appender(_doc_template)
     def resample(self, rule, *args, **kwargs):
         """
-        Provide resampling when using a TimeGrouper
-        Return a new grouper with our resampler appended
+        Provide resampling when using a TimeGrouper.
+
+        Given a grouper the function resamples it according to a string
+        "string" -> "frequency".
+
+        See the :ref:`frequency aliases <timeseries.offset-aliases>`
+        documentation for more details.
+
+        Parameters
+        ----------
+        rule : str or Offset
+            The offset string or object representing target grouper conversion.
+        *args, **kwargs
+            For compatibility with other groupby methods. See below for some
+            example parameters.
+        closed : {‘right’, ‘left’}
+            Which side of bin interval is closed.
+        label : {‘right’, ‘left’}
+            Which bin edge label to label bucket with.
+        loffset : timedelta
+            Adjust the resampled time labels.
+
+        Returns
+        -------
+        Grouper
+            Return a new grouper with our resampler appended.
+
+        Examples
+        --------
+        Start by creating a length-9 DataFrame with minute frequency.
+
+        >>> idx = pd.date_range('1/1/2000', periods=9, freq='T')
+        >>> df = pd.DataFrame(data=9 * [range(4)],
+        ...                   index=idx,
+        ...                   columns=['a', 'b', 'c', 'd'])
+        >>> df.iloc[6, 0] = 5
+        >>> df
+                             a  b  c  d
+        2000-01-01 00:00:00  0  1  2  3
+        2000-01-01 00:01:00  0  1  2  3
+        2000-01-01 00:02:00  0  1  2  3
+        2000-01-01 00:03:00  0  1  2  3
+        2000-01-01 00:04:00  0  1  2  3
+        2000-01-01 00:05:00  0  1  2  3
+        2000-01-01 00:06:00  5  1  2  3
+        2000-01-01 00:07:00  0  1  2  3
+        2000-01-01 00:08:00  0  1  2  3
```

Comment:
Do you think it'd be possible to use a more compact example. May be 4 rows of 1 minute intervals, that can be downsampled to 2 rows of 30 seconds? Also, I think 2 columns should be enough.